core/src/main/scala/kafka/common/InterBrokerSender.scala

```diff
@@ -156,6 +163,16 @@ abstract class InterBrokerSendThread(
   def wakeup(): Unit = networkClient.wakeup()
 }
 
+abstract class InterBrokerRequestManager() {
+  
+  var interBrokerSender: InterBrokerSender = _
```

Comment:
The issue is that `interBrokerSender` is being set and read in different threads. If there is no memory barrier, there is no guarantee that the reader will see the latest value being set.

Reply:
Got it. If we go back to the argument passed in as a parameter, then we can just eliminate this.

Reply:
I don't think this can be updated and read concurrently -- all updates happen in the startup thread before the object is visible to request threads.  We can add a comment to clarify.

Reply:
I'm removing this anyway in favor passing the sender when the manager is created.